contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -0,0 +1,255 @@
+ *}
+ * ```
+ *
+ * NOTE: The `delay` is only configurable in the constructor to avoid issues related with handling
+ * delay management during the transfer is pending to be completed.
+ *
+ * _Available since v4.9._
+ */
+abstract contract AccessControlDefaultAdminRules is IERC5313, AccessControl {
+    uint48 private immutable _delay;
+
+    address private _currentDefaultAdmin;
+    address private _pendingDefaultAdmin;
+
+    uint48 private _defaultAdminTransferDelayedUntil;
+
+    /**
+     * @dev Emitted when a `DEFAULT_ADMIN_ROLE` transfer is started, setting `newDefaultAdmin`
+     * as the next default admin, which will have rights to claim the `DEFAULT_ADMIN_ROLE`
+     * after `defaultAdminTransferDelayedUntil` is met.
+     */
+    event DefaultAdminRoleChangeStarted(address indexed newDefaultAdmin, uint48 defaultAdminTransferDelayedUntil);
+
+    /**
+     * @dev Sets the initial values for {delay} in seconds and {defaultAdmin}.
+     *
+     * The `delay` value is immutable. It can only be set at the constructor.
+     */
+    constructor(uint48 delay_, address initialDefaultAdmin) {
+        _delay = delay_;
+        _grantRole(DEFAULT_ADMIN_ROLE, initialDefaultAdmin);
+    }
+
+    /**
+     * @dev Returns the delay between each `DEFAULT_ADMIN_ROLE` transfer.
+     */
+    function delay() public view returns (uint48) {
+        return _delay;
+    }
+
+    /**
+     * @dev See {IERC5313-owner}.
+     */
+    function owner() public view virtual returns (address) {
+        return defaultAdmin();
+    }
+
+    /**
+     * @dev Returns the address of the current `DEFAULT_ADMIN_ROLE` holder.
+     */
+    function defaultAdmin() public view virtual returns (address) {
+        return _currentDefaultAdmin;
+    }
+
+    /**
+     * @dev Returns the address of the pending `DEFAULT_ADMIN_ROLE` holder.
+     */
+    function pendingDefaultAdmin() public view virtual returns (address) {
+        return _pendingDefaultAdmin;
+    }
+
+    /**
+     * @dev Returns the timestamp after which the pending default admin can claim the `DEFAULT_ADMIN_ROLE`.
+     */
+    function defaultAdminTransferDelayedUntil() public view virtual returns (uint48) {
+        return _defaultAdminTransferDelayedUntil;
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return
+            interfaceId == 0x4a02b518 || // type(IAccessControlDefaultAdminRules).interfaceId
+            super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @dev Starts a `DEFAULT_ADMIN_ROLE` transfer by setting a pending default admin
+     * and a timer to be met.
+     *
+     * Requirements:
+     *
+     * - Only can be called by the current `DEFAULT_ADMIN_ROLE` holder.
+     *
+     * Emits a {DefaultAdminRoleChangeStarted}.
+     */
+    function beginDefaultAdminTransfer(address newAdmin) public virtual onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
Should we have an internal function `_beginDefaultAdminTransfer` with the unprotected logic, and just call it from this `public onlyRole(DEFAULT_ADMIN_ROLE)` function?

That would allow workflows such as `beginDefaultAdminTransferWithSignature`

Reply:
Answering this and this [other comment](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009#discussion_r1112855585), I recall @frangio mentioned he was not a big fan of this pattern.

From my perspective:
1. I don't see a valuable reason to include it **now** (other use cases are definitely valuable but not sure if they're a priority)
2. Makes it more difficult to deal with overrides since now you have 2 (although this may add value)
3. I like `beginDefaultAdminTransferWithSignature` but I don't want to rush it and may add complexity to this PR

If we decide to eventually pursue `beginDefaultAdminTransferWithSignature`, let's move separate them :p

Reply:
We agreed in our weekly to separate the `public` from the `internal` implementation of both `beginDefaultAdminTransfer` and `acceptDefaultAdminTransfer`.

[Implemented in this commit](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4009/commits/8747bf4863fe4dd3cdee0904888315d6f5552fed) and solving since the recommendation was taken.